src/plots/waterfallPlot.ts

```diff
@@ -0,0 +1,194 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Plots {
+  export class Waterfall<X, Y> extends Bar<X, Y> {
+    private static _BAR_DECLINE_CLASS = "waterfall-decline";
+    private static _BAR_GROWTH_CLASS = "waterfall-growth";
+    private static _BAR_TOTAL_CLASS = "waterfall-total";
+    private static _CONNECTOR_CLASS = "connector";
+    private static _CONNECTOR_AREA_CLASS = "connector-area";
+    private static _TOTAL_KEY = "total";
+    private _connectorArea: d3.Selection<void>;
+    private _connectorsEnabled = false;
+    private _extent: number[];
+    private _subtotals: number[];
+
+    constructor() {
+      super();
+      this.addClass("waterfall-plot");
+    }
+
+    /**
+     * Gets whether connectors are enabled.
+     * 
+     * @returns {boolean} Whether connectors should be shown or not.
+     */
+    public connectorsEnabled(): boolean;
+    /**
+     * Sets whether connectors are enabled.
+     * 
+     * @param {boolean} enabled
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public connectorsEnabled(enabled: boolean): Waterfall<X, Y>;
+    public connectorsEnabled(enabled?: boolean): any {
+      if (enabled === undefined) {
+        return this._connectorsEnabled;
+      }
+      this._connectorsEnabled = enabled;
+      return this;
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for whether a bar represents a total or a delta.
+     */
+    public total<T>(): Plots.AccessorScaleBinding<T, boolean>;
+    /**
+     * Sets total to a constant number or the result of an Accessor
+     * 
+     * @param {boolean|Accessor<number>}
+     * @returns {Waterfall} The calling Waterfall Plot.
+     */
+    public total(total: boolean | Accessor<boolean>): Waterfall<X, Y>;
```

Comment:
Is there any value in having this take a `boolean`? If `true` is passed in, `Waterfall` just becomes a `Bar` `Plot`, and if `false` is passed in, I'm not sure it even works.

Related: what happens if the `Accessor` evaluates to `false` on the first element?